dlg/dropmake/web/lg_web.py

```diff
@@ -343,6 +351,81 @@ def gen_pgt():
         trace_msg = traceback.format_exc()
         return "Graph partition exception {1}: {0}".format(trace_msg, lg_name)
 
+@route('/gen_pgt', method='POST')
+def gen_pgt_post():
+    """
+    Translating Logical Graphs to Physical Graphs.
+    Differs from get_pgt above by the fact that the logical graph data is POSTed
+    to this route in a HTTP form, whereas gen_pgt loads the logical graph data
+    from a local file
+    """
+    # Retrieve the graph name.
+    lg_name = request.forms.get('lg_name')
+
+    # Retrieve json data.
+    json_string = request.forms.get('json_data')
+    logical_graph = json.loads(json.loads(json_string))
+
+    try:
+        # Save graph
+        new_path = save(lg_name, logical_graph)
+
+        # Unrolling LG to PGT.
```

Comment:
From this point onward the logic is the same in both methods, or almost the same (the only differ before this in how they get the contents of the logical graph, right?). So it would make sense to put the rest of the logic (bar the last bit, see below) into a separate, common function.